extensions/positron-notebook-controllers/src/notebookController.ts

```diff
@@ -392,38 +410,61 @@ function handleRuntimeMessageOutput(
 				cellOutputItems.push(vscode.NotebookCellOutputItem.text(data, mimeType));
 		}
 	}
-	return new vscode.NotebookCellOutput(cellOutputItems, { outputType });
+	const cellOutput = new vscode.NotebookCellOutput(cellOutputItems, { outputType });
+	await execution.appendOutput(cellOutput);
 }
 
 /**
  * Handle a LanguageRuntimeStream message.
  *
  * @param message Message to handle.
- * @returns Resulting cell output items.
+ * @param execution The cell execution to which the message belongs.
  */
-function handleRuntimeMessageStream(message: positron.LanguageRuntimeStream): vscode.NotebookCellOutput {
-	const cellOutputItems: vscode.NotebookCellOutputItem[] = [];
+async function handleRuntimeMessageStream(
+	message: positron.LanguageRuntimeStream,
+	execution: vscode.NotebookCellExecution,
+): Promise<void> {
+	let cellOutputItem: vscode.NotebookCellOutputItem;
 	switch (message.name) {
 		case positron.LanguageRuntimeStreamName.Stdout:
-			cellOutputItems.push(vscode.NotebookCellOutputItem.stdout(message.text));
+			cellOutputItem = vscode.NotebookCellOutputItem.stdout(message.text);
+			break;
 		case positron.LanguageRuntimeStreamName.Stderr:
-			cellOutputItems.push(vscode.NotebookCellOutputItem.stderr(message.text));
+			cellOutputItem = vscode.NotebookCellOutputItem.stderr(message.text);
+			break;
+		default:
+			log.warn(`Ignoring runtime message with unknown stream name: ${message.name}`);
+			return;
+	}
+
+	// If the last output has items of the same mime type (i.e. from the same stream: stdout/stderr),
+	// append the new item to the last output. Otherwise, create a new output.
+	const lastOutput = execution.cell.outputs.at(-1);
+	const lastOutputItems = lastOutput?.items;
+	if (lastOutputItems && lastOutputItems.every(item => item.mime === cellOutputItem.mime)) {
```

Comment:
I'm seeing plots printed 2x when running ⏬ which I think might be coming from this area; I wonder if there should be a restriction on mimetypes that are appended in the message stream? 

```
import matplotlib.pyplot as plt
import numpy as np

np.random.seed(1)
x = np.linspace(0, 8, 16)
y1 = 3 + 4 * x / 8 + np.random.uniform(0.0, 0.5, len(x))
y2 = 1 + 2 * x / 8 + np.random.uniform(0.0, 0.5, len(x))
fig, ax = plt.subplots()
ax.fill_between(x, y1, y2, alpha=0.5, linewidth=0)
ax.plot(x, (y1 + y2) / 2, linewidth=2)
ax.set(xlim=(0, 8), xticks=np.arange(1, 8), ylim=(0, 8), yticks=np.arange(1, 8))
plt.show() # also prints twice if you comment out this line
```

Reply:
Ah good catch, fixed in 4e0e47d. Could you please try again?